src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs

```diff
@@ -940,14 +932,7 @@ private void ThrowValidationException()
         static internal object UnsafeGetUninitializedObject(Type type)
         {
 #if !NET_NATIVE
-            if (type.GetTypeInfo().IsValueType)
-            {
-                  return Activator.CreateInstance(type);
-            }
-
-            const BindingFlags Flags = BindingFlags.Public | BindingFlags.Instance;
-            bool hasDefaultConstructor = s_typeHasDefaultConstructorMap.GetOrAdd(type, t => t.GetConstructor(Flags, Array.Empty<Type>()) != null);
-            return hasDefaultConstructor ? Activator.CreateInstance(type) : TryGetUninitializedObjectWithFormatterServices(type) ?? Activator.CreateInstance(type);
+            return FormatterServices.GetUninitializedObject(type) ?? Activator.CreateInstance(type);
```

Comment:
I don't think it's necessary either. The desktop implementation does not have it.

The only reason I kept it here is because the initial CoreFx implementation had this fallback, see [this line](https://github.com/dotnet/corefx/blob/7a8a34ef03b596165906bbe4076570fbf8b6b19d/src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/XmlFormatReaderGenerator.cs#L865). Therefore I thought it's safer to keep this.